flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/slowtaskdetector/ExecutionTimeBasedSlowTaskDetectorTest.java

```diff
@@ -200,6 +204,122 @@ void testFinishedTaskExceedRatioInDynamicGraph() throws Exception {
         assertThat(slowTasks).hasSize(2);
     }
 
+    @Test
+    void testBalancedInput() throws Exception {
+        final int parallelism = 3;
+        final JobVertex jobVertex1 = createNoOpVertex(parallelism);
+        final JobVertex jobVertex2 = createNoOpVertex(parallelism);
+        jobVertex2.connectNewDataSetAsInput(
+                jobVertex1, DistributionPattern.ALL_TO_ALL, ResultPartitionType.PIPELINED);
+        final ExecutionGraph executionGraph = createExecutionGraph(jobVertex1, jobVertex2);
+        final ExecutionTimeBasedSlowTaskDetector slowTaskDetector =
+                createSlowTaskDetector(0.3, 1, 0);
+
+        final ExecutionVertex ev21 =
+                executionGraph.getJobVertex(jobVertex2.getID()).getTaskVertices()[0];
+        ev21.setInputBytes(1024);
+        final ExecutionVertex ev22 =
+                executionGraph.getJobVertex(jobVertex2.getID()).getTaskVertices()[1];
+        ev22.setInputBytes(1024);
+        final ExecutionVertex ev23 =
+                executionGraph.getJobVertex(jobVertex2.getID()).getTaskVertices()[2];
+        ev23.setInputBytes(1024);
+
+        ev23.getCurrentExecutionAttempt().markFinished();
+        Thread.sleep(10);
```

Comment:
This sleep is not needed because ev22/ev23 are surely to have a longer execution time.